### DISCUSSION TITLE
Slideshow/Image gallery/Image carousel

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#6636] [Ideas]
An easy way in Material for MkDocs to define multiple images to be part of a gallery/Slideshow to cycle through would be a nice addition.

If technically possible could it be similar in behaviour like the grid cards feature.

It would allow us an easy option to display multiple images, without cluttering the entire page with it.
The only alternative to consider (From my knowledge) is the tabbed (Or blocks.tab) extension from PymdownX, which would have a similar effect, tho not with the automatic sliding and such...

**Comment by alexvoss** [comment]
[GLightbox](https://blueswen.github.io/mkdocs-glightbox/#demo) might be what you are looking for?

**Reply by Andre601** [reply]
No.

**Reply by alexvoss** [reply]
ok, expand?

**Reply by Andre601** [reply]
A gallery inside the page?
Idk what exactly needs to be explained here.

Just look at sides showing a gallery cycling through images...

**Reply by alexvoss** [reply]
> Idk what exactly needs to be explained here.

Ok, so the difference is that you want it integrated instead of a pop-up and cycle automatically? 

What options have you looked at? It seems to me that this is something that should be not too difficult to implement with customization. There is a [bit of JS for this here](https://stackoverflow.com/questions/47406240/add-image-carousel-in-markdown). Another possible starting point is [this tutorial](https://www.codepel.com/lightbox/javascript-carousel-slider-with-lightbox/)? There must be quite a few JS solutions out there that could conceivably be adapted.

I am not sure there is anything Material-specific in what you are asking? It would seem to me that a more generic plugin would be a good place for this to live. If the people behind GLightbox are not interested in tweaking what they have (I think they should) to cover this use case then it might be worth looking at but there are a lot of other things going on at the moment.

**Reply by Andre601** [reply]
There is a MkDocs plugin that implements a gallery inspired by Sphinx... Tho from what I can tell is that more about files than images (Kind of misleading naming in that case tbh).

I saw the first one you shared here and tried it, but no real success. Also, that example in particular is weird in that it requires images to be of same size, which is imo not as optimal, so I tried a different aproach where it checks for a `data-md-component="img-carousel"` attribute in an img tag.

I asked about such a thing on mkdocs, so feel free to check that one for further context and info...
https://github.com/mkdocs/mkdocs/discussions/3514

Just to be clear: What I want to achieve is displaying a single image that is being switched by other ones every x seconds, similar to how some wikis have such a functionality where they switch between different types of a render as an example...
And since I'm not experienced with JS would I not know what to tweak here... Or if I need to tweak anything else like some CSS for that matter...
